zerver/actions/message_send.py

```diff
@@ -553,8 +553,7 @@ def build_message_send_dict(
     testing code paths call it directly.
     """
     if realm is None:
-        realm = message.sender.realm
-    assert realm == message.realm
+        realm = message.realm
```

Comment:
Should we remove the parameter and `if` statement here? I think in any case it's not ideal to remove the `assert` as long as both parameters exist and could disagree. I guess I'll see if you do that in a later commit as I'm still reading.

Reply:
Added the assert back and then marked this to merge once CI passes.